frontend/app/src/App.tsx

```diff
@@ -825,15 +825,24 @@ export class App extends PureComponent<Props, State> {
 
   handlePageProfileMsg = (pageProfile: PageProfile): void => {
     const pageProfileObj = PageProfile.toObject(pageProfile)
+    let appId: string | null = null,
+      sessionId: string | null = null,
+      pythonVersion: string | null = null
+
+    if (this.sessionInfo.isSet) {
+      appId = this.sessionInfo.current.appId
+      sessionId = this.sessionInfo.current.sessionId
+      pythonVersion = this.sessionInfo.current.pythonVersion
+    }
```

Comment:
We might need to inform the data team that there will be a risk of page profiles with missing appId/sessionId/pythonVersion since it might be something that breaks some piplines/apps. Or maybe it's easier to just not send the page profile if session info is not available since it might not be very useful anyway (I believe the appId is quite important for many metric apps).

Reply:
Also maybe it makes sense to show a warning via `logWarning` in case session state is not set yet